[Lee Joon Jie] iP

LeeJoonJie opened the pull request.

Commit:
Add Gradle support

LeeJoonJie committed:
Implement Level-1:
1. Greet user
2. Echo user commands
3. Exit upon bye command

LeeJoonJie committed:
Create Task class to represent tasks.

LeeJoonJie committed:
Create TaskList class to represent list of tasks.

LeeJoonJie committed:
Implement Level 2:
1. Add and store tasks to list
2. Display list when requested

LeeJoonJie committed:
Implement Level 3:
1. Mark tasks as done

LeeJoonJie committed:
Create Deadline, Event, ToDo subclasses to represent different types of tasks.

LeeJoonJie committed:
Implement Level 4:
1.ToDo, Deadline, Event types of tasks

LeeJoonJie committed:
Automate Text UI Testing

LeeJoonJie committed:
Implement Level 5 (Error handling)
1. Empty task descriptions
2. Unrecognised commands
3. Invalid "done" commands (empty or out of range index)

LeeJoonJie committed:
Implement Level 6
1. Delete tasks
2. Error handling for deleting tasks
3. Update runtest.sh to include more tests

LeeJoonJie committed:
Use separate method to process user input in main

LeeJoonJie committed:
Implement Level 7 (save to hard disk)

* Load data from hard disk upon start up
* Update data to hard disk upon changes

LeeJoonJie committed:
Implement Level 8 (Deadline date)

Accept deadline dates in dd-MM-yyyy format

LeeJoonJie committed:
Merge branch 'branch-Level-7'

LeeJoonJie committed:
Merge branch 'branch-Level-8'

LeeJoonJie committed:
Parse date when retrieving from hard disk

There is runtime error when loading data from
hard disk due to difference in date format

Let's parse the date to the correct format when
loading the data

LeeJoonJie committed:
Create Storage class to load/save data from file

LeeJoonJie committed:
Create Parser class to parse user input

LeeJoonJie committed:
Create Ui class to handle user interaction

LeeJoonJie committed:
Change Task subclass constructor arguments

Subclasses of Task (Deadline, ToDo, Event) take
in user command Strings as arguments which are
then parsed in the class. However, user commands
are not very relevant to these classes.

Let's change the arguments to more relevant
fields. Such as 'deadline' and 'task' for Deadline.
The parsing of commands into these fields
can be done in the Parser class

LeeJoonJie committed:
Change deadline format to dd MMM yyyy

LeeJoonJie committed:
Remove redundant exceptions headers

LeeJoonJie committed:
Add JUnit tests for Parser and Deadline

LeeJoonJie committed:
Fix todo parser bug

The parser does not remove the "todo"
word when it should.

Let's fix the Parser to remove "todo".

LeeJoonJie committed:
Add JavaDocs to the classes and methods.

LeeJoonJie committed:
Tweak code to comply with coding standard.

LeeJoonJie committed:
Implement Level 9 (find feature)

Allow users to search for tasks containing
a specified keyword using "find {keyword}".

LeeJoonJie committed:
Merge branch 'branch-A-JavaDoc'

LeeJoonJie committed:
Merge tag 'A-CodingStandard'

LeeJoonJie committed:
Merge branch 'branch-Level-9'

LeeJoonJie committed:
Format "find" command output

LeeJoonJie committed:
Add more JavaDoc comments

LeeJoonJie committed:
Update shadow version to 6.0.0 in gradle

LeeJoonJie committed:
Add gradle files to vcs

LeeJoonJie committed:
Merge branch 'add-gradle-support'

# Conflicts:
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties
#	gradlew
#	gradlew.bat

ruilingk commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,35 @@
+    @Override
+    public String toString() {
+        return String.format("[D][%s] %s (by: %s)",
+                super.isDone() ? "\u2713" : "\u2717",
```
Maybe you could create a method in `Task.java` which returns the tick or cross symbols based on whether the Task is done? This might reduce the duplicate codes in the 3 different types of Task classes! 🤔

ruilingk commented on src/main/java/DukeException.java:
```diff
@@ -0,0 +1,9 @@
+/**
+ * Represents a class for all specialised Duke exceptions.
+ */
+public class DukeException extends Exception{
```
Should there be a spacing after Exception followed by the starting curly brace? 😀 
```suggestion
public class DukeException extends Exception {
```

ruilingk commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,32 @@
+/**
+ * Represents an event task at a certain time.
+ */
+public class Event extends Task{
```
Maybe you can take a look at `DukeException.java`? It is the same issue. 😅

ruilingk commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,32 @@
+    @Override
+    public String toString() {
+        return String.format("[E][%s] %s (at: %s)",
+                super.isDone() ? "\u2713" : "\u2717",
```
Maybe you could take a look at the suggestion mentioned in `Deadline.java`? Same for `ToDo.java`!

ruilingk commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,109 @@
+
+    private TaskList list;
+
+    protected Parser(){}
```
I am not sure about this, but I think there should be a spacing between closing parenthesis and starting curly brace? 
```suggestion
    protected Parser() {}
```

ruilingk commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,40 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+
```
Should there be a spacing between these imports? 🤔

ruilingk commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,25 @@
+
+    /**
+     * Checks if the task is done.
+     */
```
Should there be a @return for the boolean in the Javadocs? 😅

ruilingk commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,97 @@
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
```
Maybe you could look at `Storage.java`? It has the same issue.

ruilingk commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,97 @@
+            task.markDone();
+            System.out.printf("Hurray! %s is now done.\n", task.getTask());
+            System.out.println(task + "\n");
+            storage.saveTasks(toString());
```
I think there is an extra 4 spaces for the indentation?

ruilingk commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,97 @@
+            System.out.printf("Okay %s has been deleted.\n", task.getTask());
+            System.out.println(task);
+            System.out.println("You now have " + list.size() + " tasks.\n");
+            storage.saveTasks(toString());
```
Maybe you could take a look at the above comment? It is the same. 😃

ruilingk commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,17 @@
+public class Ui {
+
+    protected void start() {
+        System.out.println("Hi there! I'm Peanut.\nHow can I be of assistance?\n");
```
I like how your bot is personalised! 👍

ruilingk commented on src/test/java/DeadlineTest.java:
```diff
@@ -0,0 +1,39 @@
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.time.LocalDate;
```
Static imports should have precedence, followed by Java's imports, then followed by this Test import. Maybe when you run checkstyle, you would be clearer. At first, I did not know too! 😄 
```suggestion
import static org.junit.jupiter.api.Assertions.assertEquals;

import java.time.LocalDate;

import org.junit.jupiter.api.Test;
```

ruilingk commented on src/test/java/ParserTest.java:
```diff
@@ -0,0 +1,47 @@
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Maybe you could take a look at `ParserTest.java`? It has the same issue, and for the rest of the test classes too!

ruilingk commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,26 @@
+delete 2
+delete 5
+list
+bye
```
I like how comprehensive your testing is! 👏

ruilingk reviewed (commented):
In conclusion, I think your coding standard is great. Just some nits to amend, and you would be good to go! 👍

LeeJoonJie committed:
Create GUI feature with JavaFX

A GUI interface for Duke would be
more convenient and appealing.

Launch the GUI by running Launcher.java.
The commands are the same as the command
line version.

LeeJoonJie committed:
Merge branch 'branch-GUI'

LeeJoonJie committed:
Format code to follow coding style

LeeJoonJie committed:
Add java assertions to document assumptions

LeeJoonJie committed:
Edit code to comply with code quality

LeeJoonJie committed:
Merge pull request #1 from BreadPeanutButter/branch-A-Assertions

Add java assertions to document assumptions

LeeJoonJie committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/TaskList.java

LeeJoonJie committed:
Merge pull request #2 from BreadPeanutButter/branch-A-CodeQuality

Edit code to comply with code quality

LeeJoonJie committed:
Fix data.txt not found bug

LeeJoonJie committed:
Implement duplicate detection feature

Make Peanut alert user if he is adding a task
which already exists in the list. All tasks
with the same name are considered to be
potential duplicates regardless of type/date.

LeeJoonJie committed:
Merge branch 'branch-C-DetectDuplicates'

LeeJoonJie committed:
Change for-loop to a stream

LeeJoonJie committed:
Merge pull request #3 from BreadPeanutButter/branch-A-Streams

Change for-loop to a stream in TaskList

LeeJoonJie committed:
Use lambda as method parameter

LeeJoonJie committed:
Merge pull request #4 from BreadPeanutButter/branch-A-Lambdas

Use lambda as method parameter

LeeJoonJie committed:
Improve UI

LeeJoonJie committed:
Add Ui.png for product website

LeeJoonJie committed:
Update README in docs

LeeJoonJie committed:
Set theme jekyll-theme-architect

LeeJoonJie committed:
Change main class in gradle to Launcher

LeeJoonJie committed:
Remove redundant error

LeeJoonJie committed:
Change order of imports
